### PR TITLE
Exception for x=0

### PR DESCRIPTION
**Pull request by thiagom123**
Wrote an exception for when z=-1, as in that case the formula for alpha and beta doesn't work. If z=-1, then the program will assign a previously calculated result.

Also included a test for the exception.

### CODE REVIEW

File: qclib/gates/ldmcsu.py [hunk 1]
```diff
@@ -129,18 +129,36 @@ def linear_depth_mcv(
...
-        )
-        alpha_i = z.imag / (2. * np.sqrt(
-            (z.real + 1.) * (np.sqrt((z.real + 1.) / 2.) + 1.)
-        ))
-        alpha = alpha_r + 1.j * alpha_i
-        beta = x / (2. * np.sqrt(
+        # When x=0, the calculation of alpha and beta can be simplified.
+        # This simplification not only results in a simpler computation
+        # but also avoids the issue of dividing by zero that arises in
+        # the general case when z=-1.
```
Comment (adjs) [inline qclib/gates/ldmcsu.py:135]: remove this comment.

File: qclib/gates/ldmcsu.py [hunk 2]
```diff
@@ -129,18 +129,36 @@ def linear_depth_mcv(
...
+        if x == 0:
```
Comment (adjs) [inline qclib/gates/ldmcsu.py:136]: Should we include a test case with np.array([[np.e**(-1j*0.3), 0],[0, np.e**(1j*0.3)]])?

Comment (israelferrazaraujo) [reply qclib/gates/ldmcsu.py:132]: Done! `test_lcmcsu_op_for_exception_unitary_2`

File: qclib/gates/ldmcsu.py (hunk 1 above)
Comment (israelferrazaraujo) [reply qclib/gates/ldmcsu.py]: Removed.

**Review by adjs — APPROVED** [review]

**Comment by adjs** [comment]
Thanks @thiagom123 and @israelferrazaraujo !

**Pull request merged**